llvm/test/CodeGen/AMDGPU/wmma-gfx12-w32-imm.ll

```diff
@@ -112,7 +112,7 @@ bb:
 define amdgpu_ps void @test_wmma_bf16_16x16x16_bf16_imm(<8 x i16> %A, <8 x i16> %B, ptr addrspace(1) %out) {
 ; GFX12-LABEL: test_wmma_bf16_16x16x16_bf16_imm:
 ; GFX12:       ; %bb.0: ; %bb
-; GFX12-NEXT:    v_wmma_bf16_16x16x16_bf16 v[10:13], v[0:3], v[4:7], 1.0
+; GFX12-NEXT:    v_wmma_bf16_16x16x16_bf16 v[10:13], v[0:3], v[4:7], 0
```

Comment:
What has happened here again? I have not seen this downstream. I would understand a code bloat with the initialization with 0x3f80, but not outright 0 inline.

Also just as a note pulled from downstream: this shall use BF16 in the profile. As a minimum in the profile, but then also in the intrinsic. But downstream it still maintains 1.0, so something is missing.

Reply:
> Also just as a note pulled from downstream: this shall use BF16 in the profile. As a minimum in the profile, but then also in the intrinsic. But downstream it still maintains 1.0, so something is missing.

Yes. That is on the top of my list. I'll fix the downstream today.

Reply:
As for the intrinsic v_wmma_bf16_16x16x16_bf16, changing it is tricky because the same one is being used on older gens. The proposal (https://github.com/llvm/llvm-project/pull/77795#pullrequestreview-1839026480) was to add a new one with proper bf16 type for use on gfx12.

Reply:
Thanks for the information. I'll see what we can do here.

Reply:
No, the proposal was to change all the intrinsics. We only need to worry about maintaining compatibility with the C builtins for old operations

Reply:
If that is the case, I think https://github.com/llvm/llvm-project/pull/81669 can meet the requirement.